Monika After Story/game/script-greetings.rpy

```diff
@@ -3832,3 +3833,39 @@ label greeting_code_help_outro:
 
     $ mas_lockEVL("greeting_code_help", "GRE")
     return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.greeting_database,
+            eventlabel="greeting_back_from_store",
+            unlocked=True,
+            category=[store.mas_greetings.TYPE_STORE],
+        ),
+        code="GRE"
+    )
+
+label greeting_back_from_store:
+    if mas_isMoniNormal(higher=True):
+        m 1hua "Welcome back, [player]!"
+        m 1hub "Did you get what you needed from the store?"
+        m 2eua "I hope you chose healthy food if you bought any."
+        m 2hua "Let's spend some more time together~"
+
+    elif mas_isMoniUpset():
+        m 2esc "Oh, you're back?"
+        m 2tud "Did you forget your wallet?"
```

Comment:
I feel like this needs an absence length check for this line, given it can happen anytime within 4 hours of leaving

Reply:
The idea is that she's kind of upset that we're out for any amount of time while she is still upset with us. If you don't spend a lot of time outside, it's like a genuine "are you going to leave again?" statement, while if you're gone for hours, she says it because "you're out so long you may as well never be home" type statement. What do you think? Does it still need a split?

Reply:
I don't like that honestly. Upset right now is more 'neutral' if you will. Like she's not pleased, and somewhat doubtful.

Additionally, even if she's upset, she realizes you need to go out from time to time, so she'd be fine with you leaving. Also, when you read that, it comes across as literal, personally if she said that, I'd be expecting to respond yes/no.

Finally, that's more of an extreme assumption to make. It's possible the player still spends all day with her, but she's at upset. Without those checks in place, the idea you have behind it falls apart.

Reply:
Alright, so maybe a whole rewrite is in order here. Any suggestions?